core/integration/module_call_test.go

```diff
@@ -332,6 +333,43 @@ func (m *Test) Insecure(ctx context.Context, token *Secret) (string, error) {
 			require.ErrorContains(t, err, `unsupported secret arg source: "wtf"`)
 		})
 	})
+
+	t.Run("cache volume args", func(t *testing.T) {
+		t.Parallel()
+
+		volName := identity.NewID()
+
+		modGen := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work").
+			With(daggerExec("mod", "init", "--name=test", "--sdk=go")).
+			WithNewFile("main.go", dagger.ContainerWithNewFileOpts{
+				Contents: `package main
+
+import (
+	"context"
+)
+
+type Test struct {}
+
+func (m *Test) Cacher(ctx context.Context, cache *CacheVolume, val string) (string, error) {
+	return dag.Container().
+		From("` + alpineImage + `").
+		WithMountedCache("/cache", cache).
+		WithExec([]string{"sh", "-c", "echo $0 >> /cache/vals", val}).
+		WithExec([]string{"cat", "/cache/vals"}).
+		Stdout(ctx)
+}
+`,
+			})
+
+		out, err := modGen.With(daggerCall("cacher", "--cache", volName, "--val", "foo")).Stdout(ctx)
+		require.NoError(t, err)
+		require.Equal(t, "foo\n\n", out) // TODO: extra linebreak from 'dagger call'
```

Comment:
Feels a little weird to do that in tests though - these particular details matter in a CLI context.

Do you think we could get rid of the extra `\n`? I'm imagining obscure scenarios where someone is running `dagger call` and redirecting output to a file - IMO we should try to avoid adding anything extra because it's confusing where it'd be coming from.

Reply:
I think I agree with not appending the extra `\n`. The main argument against is that if we don't append an extra `\n` then a function like this:
```go
func (Foo) Fn() string {
   return "bar"
}
```

will result in the `bar` string be printed to stdout without a line break which sometimes makes it hard to discern from other output being printed to stderr or this (very mild) sort of nuisance where it mixes with the shell's next prompt:
```console
barsipsma@dagger_dev:~/repo/github.com/sipsma/dagger$
```

Probably more often than not if you return `"bar"` from a function and want to see the value you really want it to be shown with a `\n`. But also asking users to just return `"bar\n"` instead is not that much and is more consistent. So I think that makes sense. Obviously splitting microhairs here :-)

Reply:
Maybe in TUI mode (stdout is a TTY) we can ensure we print a final linebreak with a little ⏎ indicator like some shells do to avoid the awkward prompt clobbering.

It's only when stdout is redirected that IMO it should be faithful to the raw value.